### PR TITLE
GenAI Updates Part 4: View All Trial Page - Show results per trial

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
* Add `date_helper` and `datetime_helper` methods
* Add `Sub-Optimal Accuracy` and `Optimal Accuracy` to View all trials page
* Add GEVal average to View all trial page
* Rename `OptimalAndSuboptimalResultsBuilder` to `ConfusionMatrixBuilder`

## WHY
* UTC isn't a useful time for internal tool reporting so let's have a helper that reports default time of Eastern
* They are two of the most important metrics that we want to see for each trial
* Another important metric to help with grading quality of feedback
* We're no longer using the generic accuracy calculation so this only leaves ConfusionMatrix

## HOW
* Add methods to ApplicationHelper
* Simple #correct divided by total number
* Compute the average of GEVal scores over each test example
* Remove the unused code

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Internal-Tool-Organizing-Trials-into-Datasets-8dbc188caf194157a2a28395e907ddc0?pvs=4#240ce3d2c07a4c0190608ada6c063179

### What have you done to QA this feature?
Tested on staging

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES

### CODE REVIEW

File: services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/confusion_matrix_builder.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Evidence
+  module Research
+    module GenAI
+      class ConfusionMatrixBuilder < ApplicationService
```
Comment (dandrabik) [inline services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/confusion_matrix_builder.rb:6]: Good rename!

File: services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/confusion_matrix_builder.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        def initialize(llm_examples)
+          @llm_examples = llm_examples
+        end
+
+        # results is a 2x2 matrix where
+        # rows represent the ground truth (optimal or suboptimal),
+        # columns represent the model prediction (optimal or suboptimal),
+        def run
+          [[0, 0], [0, 0]].tap do |matrix|
+            llm_examples.each do |llm_example|
```
Comment (dandrabik) [inline services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/confusion_matrix_builder.rb:21]: There are a few ways to make this shorter, but if you want this row/column setup you could do this in the loop instead:
```ruby
row = llm_example.optimal? ? OPTIMAL : SUBOPTIMAL
column = llm_example.test_optimal? ? OPTIMAL : SUBOPTIMAL

matrix[row][column] += 1

File: services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/trial.rb [hunk 3]
```diff
@@ -63,6 +61,16 @@ def failed? = status == FAILED
         def running? = status == RUNNING
         def completed? = status == COMPLETED
 
+        def optimal_correct = confusion_matrix ? confusion_matrix[0][0] : 0
+        def suboptimal_correct = confusion_matrix ? confusion_matrix[1][1] : 0
+
+        def average_g_eval_score
+          scores = g_evals&.values&.flatten&.compact&.map(&:to_f)
```
Comment (dandrabik) [inline services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/trial.rb]: Consider making `scores` its own method.

**Review by dandrabik — APPROVED** [review]
Looks good. Some small refactor suggestions.

File: services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/confusion_matrix_builder.rb (hunk 2 above)
Comment (brendanshean) [reply services/QuillLMS/engines/evidence/app/services/evidence/research/gen_ai/confusion_matrix_builder.rb:21]: Thanks!  That's much cleaner.

**Pull request merged**